pkgs/desktops/deepin/dde-file-manager/default.nix

```diff
@@ -0,0 +1,235 @@
+
+    fixPath ${avfs} /usr/bin/mountavfs dde-file-manager-lib/shutil/fileutils.cpp
+    fixPath ${avfs} /usr/bin/umountavfs dde-file-manager-lib/shutil/fileutils.cpp
+
+    fixPath ${deepin-terminal} /usr/bin/deepin-terminal \
+      dde-file-manager-lib/shutil/fileutils.cpp
+
+    fixPath $out /usr/share/dde-file-manager \
+      dde-sharefiles/appbase.pri \
+      dde-sharefiles/dde-sharefiles.pro
+
+    fixPath $out /usr/share/usb-device-formatter \
+      usb-device-formatter/main.cpp
+
+    fixPath $out /usr/share/applications \
+      dde-file-manager/mips/dde-file-manager-autostart.desktop \
+      dde-desktop/development.pri
+
+    fixPath $out /usr/bin \
+      dbusservices/com.deepin.dde.desktop.service \
+      dde-desktop/data/com.deepin.dde.desktop.service \
+      dde-desktop/dbus/filedialog/com.deepin.filemanager.filedialog.service \
+      dde-desktop/dbus/filemanager1/org.freedesktop.FileManager.service \
+      dde-file-manager-daemon/dbusservice/com.deepin.filemanager.daemon.service \
+      dde-file-manager-daemon/dbusservice/dde-filemanager-daemon.service \
+      dde-file-manager-daemon/dde-file-manager-daemon.pro \
+      dde-file-manager-lib/dde-file-manager-lib.pro \
+      dde-file-manager-lib/pkexec/com.deepin.pkexec.dde-file-manager.policy \
+      dde-file-manager/dde-file-manager-xdg-autostart.desktop \
+      dde-file-manager/dde-file-manager.desktop \
+      dde-file-manager/dde-file-manager.pro \
+      dde-file-manager/mips/dde-file-manager-autostart.desktop \
+      dde-file-manager/mips/dde-file-manager.desktop \
+      dde-file-manager/pkexec/com.deepin.pkexec.dde-file-manager.policy \
+      usb-device-formatter/pkexec/com.deepin.pkexec.usb-device-formatter.policy \
+      usb-device-formatter/usb-device-formatter.desktop \
+      usb-device-formatter/usb-device-formatter.pro
+      fixPath $out /etc \
+      dde-file-manager/dde-file-manager.pro \
+      dde-file-manager-daemon/dde-file-manager-daemon.pro
+
+    fixPath $out /usr \
+      common/common.pri \
+      dde-desktop/dbus/filedialog/filedialog.pri \
+      dde-desktop/dbus/filemanager1/filemanager1.pri \
+      dde-desktop/development.pri \
+      dde-dock-plugins/disk-mount/disk-mount.pro \
+      dde-file-manager-daemon/dde-file-manager-daemon.pro \
+      usb-device-formatter/usb-device-formatter.pro
+
+    sed -i -e "s,Exec=dde-file-manager,Exec=$out/bin/dde-file-manager," \
+      dde-file-manager/dde-file-manager.desktop
+
+    sed -i -e "s,Exec=gio,Exec=${glib.bin}/bin/gio," \
+      dde-desktop/data/applications/dde-trash.desktop \
+      dde-desktop/data/applications/dde-computer.desktop
+
+    sed -i -e "s,/usr/lib/gvfs/gvfsd,${gvfs}/libexec/gvfsd," \
+      dde-file-manager-lib/gvfs/networkmanager.cpp
+
+    # I do not know yet how to fix the ocurrence of QString("/etc/xdg/%1/%2") in 
+    #   dde-file-manager-lib/interfaces/dfmsettings.cpp
+    # Should it be fixed somehow, or can it stay as it is?
```

Comment:
Probably needs another patch to load things from `XDG_CONFIG_DIRS` I think.

Reply:
> Probably needs another patch to load things from `XDG_CONFIG_DIRS` I think.

Looking at the source code it can be seen that all the standard locations are tried before faling back to `/etc/xdg`: https://github.com/linuxdeepin/dde-file-manager/blob/master/dde-file-manager-lib/interfaces/dfmsettings.cpp#L298

``` c++
    const QStringList &list = QStandardPaths::standardLocations(type);

    QString path =
       list.isEmpty()
          ? QString("/etc/xdg/%1 %2").arg(qApp->organizationName()).arg(qApp->applicationName())
          : list.last();
```

Therefore this does not need a fix.

Reply:
Great.